### PR TITLE
[Serve] Collect telemetry data for multiplexed api

### PR DESCRIPTION
**Pull request by sihanwang41**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Review by GeneDer — APPROVED** [review]
LGTM

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
telemetry test failing

**Review by pcmoritz — APPROVED** [review]

**Review by edoakes — APPROVED** [review]

**Comment by edoakes** [comment]
FYI @sihanwang41 I'm refactoring the usage code a bit here: https://github.com/ray-project/ray/pull/38658

One of us will need to deal with the conflicts :)

File: python/ray/serve/tests/test_telemetry.py [hunk 1]
```diff
@@ -640,5 +644,52 @@ def check_telemetry(tag_should_be_set: bool):
         wait_for_condition(check_telemetry, tag_should_be_set=True)
 
 
+def test_multiplexed_detect(manage_ray):
+    """Check that multiplexed api is detected by telemetry."""
+
+    subprocess.check_output(["ray", "start", "--head"])
+    wait_for_condition(check_ray_started, timeout=5)
+
```
Comment (edoakes) [inline python/ray/serve/tests/test_telemetry.py:652]: please add a check here that the multiplexed usage tag is _not_ present prior to using it

Comment (sihanwang41) [reply python/ray/serve/tests/test_telemetry.py:652]: added

**Pull request merged**